tests/helpers/class-wc-helper-subscription.php

```diff
@@ -10,19 +10,34 @@
  *
  * This helper class should ONLY be used for unit tests!.
  */
-class WC_Subscription {
+class WC_Subscription extends WC_Order {
```

Comment:
It's unclear what is the need for extending WC_Order, could you please share your thoughts?

Reply:
Just trying to mimic the real [WC_Subscription](https://github.com/woocommerce/woocommerce-subscriptions/blob/trunk/includes/class-wc-subscription.php#L15).

Also, in my unit test, I try to do what [update_payment_method()](1148-gh-woocommerce/woocommerce-subscriptions/blob/trunk/includes/class-wc-subscriptions-change-payment-gateway.php#L497) does and there are calls to WC_Order methods like [update_meta_data()](https://github.com/Automattic/woocommerce-payments/pull/1148/files#diff-933977839629fb49196cf683653de08b29c2f319c0cc0edb91ecac15d1e747a6R205).
Now it makes me wonder, if that call would make database change and if so, how does that happen in unit test.

Reply:
You asked yourself a good question. If you take a look at other helper classes they have fake/mock implementations. Inheriting from production `WC_Order` is not correct approach.

Reply:
I have removed this part. I didn't finish the unit test part. Please help to finish the unit test, @dmvrtx. Thank you!

Reply:
Interestingly, I see the test cases creating `WC_Order` istances as needed and then working with `update_meta_data()`.

Reply:
I found a mock class for `WC_Data` under WooCommerce repo. For now I copied it under the `tests/helpers` path but I wonder if there is a better way to re-use it.